File: backend/config-services/src/main/java/com/microservices/configservices/service/TopicService.java
package com.microservices.configservices.service;

import java.util.List;
import java.util.UUID;
import com.microservices.configservices.payload.request.TopicRequest;
import com.microservices.configservices.payload.response.TopicResponse;

public interface TopicService {

    TopicResponse create(TopicRequest topicRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    TopicResponse getById(UUID id);

    List<TopicResponse> getAll();

    TopicResponse update(UUID id, TopicRequest topicRequest);

    TopicResponse delete(UUID id);
}

File: frontend/reactjs-app/src/services/TopicService.js
import httpAxios from '../httpAxios';

const Topicservice = {
    create: (topicData) => {
        return httpAxios.post('config-services/api/topics/create', topicData);
    },
    setImage: (data) => {
        return httpAxios.put('config-services/api/topics/set-image', data);
    },
    getById: (id) => {
        return httpAxios.get(`config-services/api/topics/get-by-id/${id}`);
    },
    getAll: () => {
        return httpAxios.get('config-services/api/topics/get-all');
    },
    update: (id, topic) => {
        return httpAxios.put(`config-services/api/topics/update/${id}`, topic);
    },
    switchStatus: (id) => {
        return httpAxios.put(`config-services/api/topics/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`config-services/api/topics/trash/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`config-services/api/topics/delete/${id}`);
    },
};

export default Topicservice;
